src/support.js

```diff
@@ -671,18 +741,25 @@ function addFunction(func, sig) {
 #endif
 
 #endif // EMULATED_FUNCTION_POINTERS == 0
+
+#endif // WASM_BACKEND
 }
 
 function removeFunction(index) {
+#if WASM_BACKEND
```

Comment:
how about
```
#if WASM_BACKEND
  return; // nothing to do here
#endif
```
? Easier to see it doesn't do anything here in that case.

Should it really do nothing, though? Reusing indexes may be nice.

Reply:
Since indexes here are effectively function pointers, I'm not sure it makes much sense to re-use.   Having said that i suppose its conceivable that this would happen in native code with "dlopen() + dlsym() dlclose()" called a in cycle you might end up getting a function pointer that aliased a previous out?

Right now it makes sense to do nothing though, no least because we don't any "nullref" that we could put into the slot.  We'd also be forece into a more complex lookup for free slots.

BTW, do you know how often `addFunction` is used in practice?  In the whole of emscripten we only have this single test case that tinkles its so I'm guessing its rare.

Reply:
Probably pretty rare, since dlsym uses addWasmFunction, which is valid if the other module is wasm. But e.g. using the WebIDL binder you can mix JS and wasm in interesting ways that require addFunction.

Reply:
Added TODO